### ISSUE TITLE
[RFC] Add a make:form:type-extension command

### ISSUE THREAD

**Issue by yceruto** [symfony/maker-bundle#78]
Hi guys!

This maker command would generates a new (form) type extension class under `src/Form/Extension/` and the argument is the extended type class (the class name is derived from it).

What do you think?

**Comment by javiereguiluz** [comment]
I guess it's OK to add this maker. About the naming, we need to carefully think about this (not only for this proposal, but for others).

At first we thought we could use this syntax: `make : xxx` (e.g. `make:twig-extension`) to avoid the boring triple namespace (`make:twig:extension`). It's great because it's short. And for cases like Twig, it's the best option because we won't generate anything related to Twig except the extension, so there won't be more `make:twig:xxx`.

However, for things like `auth`, we are planning to do `make:auth:xxx` so maybe we cannot escape from the triple namespace curse. In summary, we already have `make:form`; if we add this command, what's the best name for it?

```
make:form:extension
make:form-extension
```

I'd remove the `type-` part in all cases because in Symfony Forms, extensions can only be about "types", so it's redundant.

**Comment by stof** [comment]
> I'd remove the type- part in all cases because in Symfony Forms, extensions can only be about "types", so it's redundant.

that's not true. The component has both a FormExtension and a FormTypeExtension (but the FormExtension is not that used, as we rely on the DependencyInjectionExtension in any project using FrameworkBundle, and we configure this extension instead).

**Comment by stof** [comment]
Regarding the naming, I don't think the type extension should be named based on the type it extends. You can have many type extensions extending the same type.
Naming them based on the feature they add is much better.

**Comment by yceruto** [comment]
> You can have many type extensions extending the same type. Naming them based on the feature they add is much better.

@stof Good point! I agree.

So, it should be named `make:form-type-extension`?

**Comment by yceruto** [comment]
See https://github.com/symfony/maker-bundle/pull/85

**Comment by gabiudrescu** [comment]
any chance this can be revived?

If I provide an implementation based on the initial implementation from @yceruto can this be merged to master?

**Comment by weaverryan** [comment]
I would be 👍 for having this maker :) The original PR was just SO early in Maker dev, that we changed a bunch of stuff and broke that PR :/.
